contracts/NameWrapper.sol

```diff
@@ -561,37 +568,33 @@ contract NameWrapper is
 
     function _mint(
         bytes32 node,
-        bytes memory name,
         address wrappedOwner,
         uint96 _fuses
-    ) internal {
-        names[node] = name;
-
+    ) internal override {
         address oldWrappedOwner = ownerOf(uint256(node));
         if (oldWrappedOwner != address(0)) {
+            // sanity check that a new token isn't owned by the wrapper
+            assert(oldWrappedOwner != address(this));
             // burn and unwrap old token of old owner
             _burn(uint256(node));
             emit NameUnwrapped(node, address(0));
         }
-        _mint(node, wrappedOwner, _fuses);
+        super._mint(node, wrappedOwner, _fuses);
     }
 
     function _wrap(
         bytes memory name,
+        bytes32 node,
```

Comment:
We can avoid this if we move the event out of _wrap, but it means we'd have to add emit Event everywhere. Alternatively we could remove the node from the event, since that could be derived from the name.